### DISCUSSION TITLE
Images - Wrong approach?

### DISCUSSION THREAD

**Discussion by jenstornell** [directus/directus#2857] [Archive]
In Directus it's possible to add images which is great. However I don't really like the approach.

### Downsides with the current approach

The images are stored in Directus own tables. That makes us dependent of Directus. The reason I like Directus in the first place is that my data is separated from Directus.

### Solution

- When using a file field in a table field, show a file browser, kind of like my mockup below.
- When setting up the file field in the options, set a rootpath (defaults to project root path).
- When setting up the file field in the options, set a prefix that will be saved before the path (optional).
- Allow upload to the folder OR pick an already uploaded file.

![image](https://user-images.githubusercontent.com/927011/94939540-971f8280-04d2-11eb-932a-4855880c2439.png)

### Gains with the new approach

- Not dependent of Directus.
- The file tree can look however the user want. It does not need to be flat.
- We can use any filenames we want and keep them that way.
- Multiple Directus setups can use the same images.
- It's easier to have total control of the filepath in the database as well as the file organization.

The only downside I see is that it may take a bit of time to build a file browser and there may be a future need for delete, rename files and folders as well.

**Comment by benhaynes** [comment]
Thanks @jenstornell — you might want to add this as a Discussion on the v9 repo so it doesn't get overlooked. We're about to transition from v8 to v9 in the next few weeks.

https://github.com/directus/next/discussions
